skyportal/handlers/api/source.py

```diff
@@ -606,13 +607,33 @@ async def get_sources(
             followup_request_subquery, Obj.id == followup_request_subquery.c.obj_id
         )
     if has_been_labelled:
-        labels_subquery = SourceLabel.select(session.user_or_token).subquery()
-        obj_query = obj_query.join(labels_subquery, Obj.id == labels_subquery.c.obj_id)
+        if current_user_labeller:
+            labels_subquery = (
+                SourceLabel.select(session.user_or_token)
+                .where(SourceLabel.labeller_id == user.id)
+                .subquery()
+            )
+            obj_query = obj_query.join(
+                labels_subquery, Obj.id == labels_subquery.c.obj_id
+            )
+        else:
+            labels_subquery = SourceLabel.select(session.user_or_token).subquery()
+            obj_query = obj_query.join(
+                labels_subquery, Obj.id == labels_subquery.c.obj_id
+            )
     if has_not_been_labelled:
-        labels_subquery = SourceLabel.select(
-            session.user_or_token, columns=[SourceLabel.obj_id]
-        ).subquery()
-        obj_query = obj_query.where(Obj.id.notin_(labels_subquery))
+        if current_user_labeller:
+            labels_subquery = (
+                SourceLabel.select(session.user_or_token, columns=[SourceLabel.obj_id])
+                .where(SourceLabel.labeller_id == user.id)
+                .subquery()
+            )
+            obj_query = obj_query.where(Obj.id.notin_(labels_subquery))
+        else:
+            labels_subquery = SourceLabel.select(
+                session.user_or_token, columns=[SourceLabel.obj_id]
+            ).subquery()
+            obj_query = obj_query.where(Obj.id.notin_(labels_subquery))
```

Comment:
same as above. In fact, you can probably get the `if current_user_labeller` conditional after of the has_been/has_not_been conditionals.

Reply:
I restructured the conditionals and tested to make sure the checkboxes still work as intended.